src/unicodedb/types.nim

```diff
@@ -21,10 +21,42 @@ proc unicodeTypes*(cp: Rune): int {.inline.} =
   ## Return types for a given code point.
   ## Use `contains` to retrieve a single type
   assert cp.int <= 0x10FFFF
-  let
-    blockOffset = (typesOffsets[cp.int div blockSize]).int * blockSize
-    idx = typesIndices[blockOffset + cp.int mod blockSize]
-  result = typesData[idx]
+  when nimvm:
+    #[
+    ugly workaround for https://github.com/nitely/nim-regex/issues/4
+    typesOffsets.len = 8704
+    typesIndices.len = 31104 ; this gives a code size of ~93321 instructions
+    (3 instructions per array element), which doesn't fit in `int16.high`
+    required in vmgen (0x7fff)
+    ]#
+    const N = typesIndices.len
+    const N2 = N div 3
+    const t0 = typesIndices[0 ..< N2]
+    const t1 = typesIndices[N2 ..< 2*N2]
+    const t2 = typesIndices[2*N2 ..< N]
+
+    proc getTypeIndex(sub: static int, ind: int): auto =
```

Comment:
change it to `static[int]` to make the tests green. I should drop Nim 0.18 support at some point...

Reply:
done!